solutions/usaco-767.mdx

```diff
@@ -0,0 +1,60 @@
+---
+id: usaco-767
+source: USACO Gold 2017 December
+title: Haybale Feast
+author: Juheon Rhee
+---
+
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_hayfeast_gold_dec17.html)
+
+## Implementation
+
+<LanguageSection>
+
+<PySection>
+
+```python
+import sys
+
+sys.stdin = open("hayfeast.in", "r")
+sys.stdout = open("hayfeast.out", "w")
+
+input = sys.stdin.readline
+
+n, m = map(int, input().split())
+
+put = [p for p in range(n)]
+
+way = [[*map(int, input().split())] + [i] for i in range(n)]
+
+sorted_way, res = sorted(way, key = lambda a: a[1]), [0] * n
+
+for i in range(n):
+        res[i] = way[i][0]
+
+def search(x):
+        if put[x] == x:
+                return x
+        put[x] = search(put[x])
+        return put[x]
+
+def add(x, y):
```

Comment:
take out of functions if possible, i only see this being used once

Reply:
i actually think its more comprehensive when its inside a function

Reply:
how

Reply:
bc u can tell i used an algorithm

Reply:
wha

Reply:
bump